src/components/event/EventItem/Event-item.js

```diff
@@ -24,166 +24,153 @@ import eventDefaultImage from "../../../constants/eventDefaultImage";
 const useStyles = useStyle;
 const photoService = new PhotoService();
 
-export default class EventCard extends Component {
-  constructor(props) {
-    super(props);
+const EventCard = props => {
+  const { item, onBlock, onUnBlock } = props;
+  const [eventImage, setEventImage] = useState(eventDefaultImage);
 
-    this.state = {
-      eventImage: eventDefaultImage,
+  useEffect(() => {
+    photoService.getPreviewEventPhoto(item.id).then(eventPreviewImage => {
+      if (eventPreviewImage) {
+        setEventImage(URL.createObjectURL(eventPreviewImage));
+      }
+    });
+    return () => {
+      URL.revokeObjectURL(eventImage);
     };
-  }
+  });
 
-  componentDidMount() {
-    photoService
-      .getPreviewEventPhoto(this.props.item.id)
-      .then(eventPreviewImage => {
-        if (eventPreviewImage != null) {
-          this.setState({ eventImage: URL.createObjectURL(eventPreviewImage) });
-        }
-      });
-  }
-
-  componentWillUnmount() {
-    URL.revokeObjectURL(this.state.eventImage);
-  }
-
-  renderCategories = arr => {
+  const renderCategories = arr => {
     return arr.map(x => <span key={x.id}>#{x.name}</span>);
   };
+  const classes = useStyles;
+  const {
+    id,
+    title,
+    dateFrom,
+    description,
+    isPublic,
+    isOnlyForAdults,
+    maxParticipants,
+    eventStatus,
+    categories,
+    countVisitor,
+    organizers: owners,
+    members,
+  } = item;
+  const INT32_MAX_VALUE = null;
```

Comment:
Please remove this variable, because I don't think it needed. This variable uses on 92 line, but it always checks for null, and it will always pass

Reply:
Removed variable